# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/QuotaParent.cpp
# Commit: b12a2f55233f
# Full Hash: b12a2f55233fade8f2e3411295ff3d362df9e359
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-09 21:46:09
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Change nsIQuotaManagerService::ResetStoragesForPrincipal to support resetting of entire origins only; r=dom-storage-reviewers,jari
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195510
# ==============================================================================

diff -r 6a90f36c8bf5 -r b12a2f55233f dom/quota/QuotaParent.cpp
--- a/dom/quota/QuotaParent.cpp	Wed Oct 09 11:59:34 2024 +0000
+++ b/dom/quota/QuotaParent.cpp	Wed Oct 09 11:59:35 2024 +0000
@@ -755,7 +755,7 @@
 
 mozilla::ipc::IPCResult Quota::RecvShutdownStoragesForOrigin(
     const Maybe<PersistenceType>& aPersistenceType,
-    const PrincipalInfo& aPrincipalInfo, const Maybe<Type>& aClientType,
+    const PrincipalInfo& aPrincipalInfo,
     ShutdownStoragesForOriginResolver&& aResolver) {
   AssertIsOnBackgroundThread();
 
@@ -770,19 +770,13 @@
 
     QM_TRY(MOZ_TO_RESULT(QuotaManager::IsPrincipalInfoValid(aPrincipalInfo)),
            QM_CUF_AND_IPC_FAIL(this));
-
-    if (aClientType) {
-      QM_TRY(MOZ_TO_RESULT(Client::IsValidType(*aClientType)),
-             QM_CUF_AND_IPC_FAIL(this));
-    }
   }
 
   QM_TRY_UNWRAP(const NotNull<RefPtr<QuotaManager>> quotaManager,
                 QuotaManager::GetOrCreate(),
                 ResolveBoolResponseAndReturn(aResolver));
 
-  quotaManager
-      ->ShutdownStoragesForOrigin(aPersistenceType, aPrincipalInfo, aClientType)
+  quotaManager->ShutdownStoragesForOrigin(aPersistenceType, aPrincipalInfo)
       ->Then(GetCurrentSerialEventTarget(), __func__,
              BoolPromiseResolveOrRejectCallback(this, std::move(aResolver)));
 